internal/controllers/topology/cluster/patches/variables/variables.go

```diff
@@ -64,6 +67,20 @@ type ClusterTopologyBuiltins struct {
 	Class string `json:"class,omitempty"`
 }
 
+// ClusterNetworkBuiltins represents builtin cluster network variables.
+type ClusterNetworkBuiltins struct {
+	// APIServerPort is the port the API Server should bind to.
+	APIServerPort *int32 `json:"apiServerPort,omitempty"`
+	// 	ServiceDomain is the network ranges from which service VIPs are allocated.
+	ServiceDomain string `json:"serviceDomain,omitempty"`
```

Comment:
```suggestion
	ServiceDomain *string `json:"serviceDomain,omitempty"`
```
I think this should be a pointer too. Given that the Cluster variable is also option (don't ask me why the serviceDomain in the Cluster is not a pointer :) )